docs/internal/types.md

```diff
@@ -473,3 +477,23 @@ class MyClass : EqualComparable, Addable
 ```
 
 In the example above, the `MyClass` argument type _matches_ the `Self` type in the interface declaration because it is the same type as the class declaring it.
+
+If instead we had to use the interface's name, the methods `==` and `+` would match _any_ types that implement the interfaces.
+
+Example:
+```ts
+interface Addable
+{
+  +(Addable & readonly, Addable & readonly) : Addable & imm;
+}
+class Foo : Addable { ... }
+class Bar : Addable { ... }
+class Baz : Addable { ... }
```

Comment:
Explicitly stating that you conform to an interface that you don't conform to is a bit confusing.  It would help if `Foo` implemented `+(Foo, Foo) : Foo` to show that this doesn't match `+(Addable, Addable) : Addable`.

Reply:
That was implied by the three dots, I didn't want to bloat the example. But I can add if you think that's clearer.